### PR TITLE
feat(api): linalg

### PR DESCRIPTION
**Pull request by dcvz**

### CODE REVIEW

**Comment by minghuaw** [comment]
@dcvz could you please remind me why this was closed?

**Comment by dcvz** [comment]
@minghuaw no big reason, it was just pending some other functionality and closed it in the meantime. Plan was to continue from here but didn’t wanna block it as “mine”.

**Comment by minghuaw** [comment]
Re-opening to continue working on linalgs

**Comment by minghuaw** [comment]
### Dependencies

- [ ] #76

File: src/linalg.rs [hunk 1]
```diff
@@ -0,0 +1,696 @@
...
+/// let (u, s, vt) = unsafe { svd_device_unchecked(&a, StreamOrDevice::cpu()) };
+/// let u_expected = Array::from_slice(&[-0.404554, 0.914514, -0.914514, -0.404554], &[2, 2]);
+/// let s_expected = Array::from_slice(&[5.46499, 0.365966], &[2]);
+/// let vt_expected = Array::from_slice(&[-0.576048, -0.817416, -0.817415, 0.576048], &[2, 2]);
+/// assert!(u.all_close(&u_expected, None, None, None).unwrap().item::<bool>());
+/// assert!(s.all_close(&s_expected, None, None, None).unwrap().item::<bool>());
+/// assert!(vt.all_close(&vt_expected, None, None, None).unwrap().item::<bool>());
+/// ```
+#[default_device]
+pub unsafe fn svd_device_unchecked(
```
Comment (dcvz) [inline src/linalg.rs:300]: `unchecked_` on purpose?

File: src/linalg.rs [hunk 2]
```diff
@@ -0,0 +1,696 @@
...
+    fn into_option(self) -> Option<Ord<'a>> {
+        Some(Ord::P(self))
+    }
+}
+
+#[default_device]
+pub fn norm_p_device<'a>(
+    array: &Array,
+    ord: f64,
+    axes: impl Into<Option<&'a [i32]>>, // TODO: use IntoOption
```
Comment (dcvz) [inline src/linalg.rs:57]: This could be resolved now, right?

**Review by dcvz — COMMENTED** [review]
LGTM, two small comments though. Feel free to resolve and merge.

File: src/linalg.rs (hunk 1 above)
Comment (minghuaw) [reply src/linalg.rs:300]: Ahh, forgot to remove this

File: src/linalg.rs (hunk 2 above)
Comment (minghuaw) [reply src/linalg.rs]: Resolved in [6202a12](https://github.com/oxideai/mlx-rs/pull/51/commits/6202a129a0267d7b45dcfd1b6fe6162934480eed)

File: src/linalg.rs (hunk 1 above)
Comment (minghuaw) [reply src/linalg.rs]: Resolved in [6202a12](https://github.com/oxideai/mlx-rs/pull/51/commits/6202a129a0267d7b45dcfd1b6fe6162934480eed)

**Pull request merged**